api/v1alpha1/kafkauser_types.go

```diff
@@ -33,6 +33,8 @@ type KafkaUserSpec struct {
 	IncludeJKS     bool              `json:"includeJKS,omitempty"`
 	CreateCert     *bool             `json:"createCert,omitempty"`
 	PKIBackendSpec *PKIBackendSpec   `json:"pkiBackendSpec,omitempty"`
+	// Duration defines the duration of the validity of the generated certificate for user
+	Duration *metav1.Duration `json:"duration,omitempty"`
```

Comment:
I would create a getter function and hardcoded the 90 day validation for cert-manager and k8s-csr case.
like: 
```
func (spec *KafkaUserSpec) GetValidationSecond() int32 {
	if spec.ValidationSecond == nil {
		return 90 * 24 * 3600
	}
	return spec.ValidationSecond
}
```

Reply:
> I think it would be better to use this one:
> 
> ```
> // expirationSeconds is the requested duration of validity of the issued certificate.
> // The minimum valid value for expirationSeconds is 3600 i.e. 1h.
> // When it is not specified the default validation duration is 90 days
> // +optional
> // +kubebuilder:validation:Minimum=3600
> ExpirationSeconds *int32 `json:"expirationSeconds,omitempty" 
> ```
> 
> Same used here: https://github.com/kubernetes/kubernetes/blob/7f6f68fdabc4df88cfea2dcf9a19b2b830f1e647/pkg/apis/certificates/types.go#L91
> 
> I think to use the ExpirationSeconds name is more descriptive.

Here, there are 2 possibilities, for cert manager minimum time is 1 hour and for k8s-csr it is 10 mins, and for cert-manager it is *metav1.Duration and for k8s-csr it is taking *int432. Which one we should provide?

Reply:
> I would create a getter function and hardcoded the 90 day validation for cert-manager and k8s-csr case. like:
> 
> ```
> func (spec *KafkaUserSpec) GetValidationSecond() int32 {
> 	if spec.ValidationSecond == nil {
> 		return 90 * 24 * 3600
> 	}
> 	return *spec.ValidationSecond
> }
> ```
> 
> In this way we can be certain that the certificate is valid for 90 days when it is not specified

ok I will add it.

Reply:
"Here, there are 2 possibilities, for cert manager minimum time is 1 hour and for k8s-csr it is 10 mins, and for cert-manager it is *metav1.Duration and for k8s-csr it is taking *int32. Which one we should provide?"

Good question. My first idea was to go with minimum 1h and when somebody really need to go under 1h duration for k8s-csr we can make another PR. 
- Im also thinking on that solution that we go with minimum 10min and write in the description about the minimum 1h for cert-manager. 
In the codebase when the duration is under 1h for cert-manager we should return error.

- Other solution (best) would be to create a validation webhook for kafkauser thus we can check both case with different value (k8s-csr and cert-manager) I think this is not worth the effort.

I think it is better to go with the minimum 1h because most of the user will not read description and I think it is an edge case to go under 1h. 
What others think?

Reply:
I agree with going with a 1 hour minimum, I can't really think of anything right now that would make that period too long, but if needed it can be modified later on for sure.

I also think it's best to have a universal limit instead of an edge-case hidden somewhere in the docs.

Reply:
Done